### PR TITLE
Remove service telemetry

### PR DESCRIPTION
**Pull request by Dexterp37**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Review by brizental — APPROVED** [review]

File: .buildconfig.yml [hunk 1]
```diff
@@ -301,10 +301,6 @@ projects:
 #  service-pocket:
 #    path: components/service/pocket
 #    description: 'A library to communicate with the Pocket API'
-#    publish: true
-#  service-telemetry:
-#    path: components/service/telemetry
-#    description: 'A generic library for generating and sending telemetry pings from Android applications to the Mozilla telemetry service.'
```
Comment (pocmo) [inline .buildconfig.yml:307]: This is already back commented in, you may need to rebase before you can merge. :)

File: docs/changelog.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ permalink: /changelog/
 
 * **feature-recentlyclosed**
   * Added a new [RecentlyClosedTabsStorage] and a [RecentlyClosedMiddleware] to maintain a list of restorable recently closed tabs.
+* **service-telemetry**
+  * This library has been removed. Please use `service-glean` instead.
```
Comment (pocmo) [inline docs/changelog.md:18]: `master` is on 59.0 now.

**Review by pocmo — APPROVED** [review]

**Pull request merged**